server/konnectors/edf.coffee

```diff
@@ -0,0 +1,1358 @@
+                entries.consumptionStatements = entries
+                .consumptionStatements.concat statements
+
+            K.logger.info 'Fetched fetchEdeliaMonthlyGasConsumptions'
+
+        catch e
+            error = e
+        finally
+            callback error
+
+fetchEdeliaSimilarHomeYearlyGasComparisions = (
+requiredFields, entries, data, callback) ->
+    return callback() if data.noEdelia or data.noGas
+
+    K.logger.info "fetchEdeliaSimilarHomeYearlyGasComparisions"
+    getEdelia data.edeliaToken
+    , "/sites/-/similar-home-yearly-gas-comparisons?begin-year=2012"
+    , (err, response, objs) ->
+
+        error = null
+        try
+            if response.statusCode is 404 or response.statusCode is 500
+                K.logger.warn 'No EdeliaSimilarHomeYearlyGasComparisions'
+                throw null
+
+            if err
+                K.logger.error(
+                    'While fetchEdeliaSimilarHomeYearlyGasComparisions')
+                K.logger.error err
+                throw err
+
+            objs.forEach (obj) ->
+                # if ...
+                statement = data.consumptionStatementByYear[obj.year]
+                statement.similarHomes =
+                    site: obj.energies.site
+                    average: obj.energies.similarHomes.SH_AVERAGE_CONSUMING
+                    least: obj.energies.similarHomes.SH_LEAST_CONSUMING
+
+            K.logger.info 'Fetched fetchEdeliaSimilarHomeYearlyGasComparisions'
+        catch e
+            error = e
+        finally
+            callback error
+        callback()
+
+fetchEdeliaGasIndexes = (requiredFields, entries, data, callback) ->
+    return callback() if data.noEdelia or data.noGas
+
+    K.logger.info "fetchEdeliaGasIndexes"
+    getEdelia data.edeliaToken
+    , "/sites/-/gas-indexes?begin-date=2012-01-01&" +
+    "end-date=#{moment().format('YYYY-MM-DD')}&types="
+    , (err, response, objs) ->
+
+        error = null
+        try
+            if response.statusCode is 404
+                K.logger.warn 'No EdeliaGasIndexes'
+                throw null
+
+            if err
+                K.logger.error 'Wihle fetchEdeliaGasIndexes'
+                K.logger.error err
+                throw callback err
+
+            objs.forEach (obj) ->
+                statement = data
+                    .consumptionStatementByMonth[obj.date.slice(0, 7)]
+
+                statement.releve = statement.releve || []
+                statement.releve.push obj
+
+            K.logger.info 'Fetched fetchEdeliaGasIndexes'
+        catch e
+            error = e
+        finally
+            callback error
+        callback()
+
+##
+
+
+prepareEntries = (requiredFields, entries, data, next) ->
+    entries.homes = []
+    entries.consumptionStatements = []
+    entries.contracts = []
+    entries.bills = []
+    # entries.clients = []
+    # entries.paymentTerms = []
+    next()
+
+updateOrCreateDocs = (requiredFields, entries, data, next) ->
```

Comment:
No, I can't use filter_existing because:
1. I need custom criteria on filtering, specific for each DocType
2. I need updateOrCreate, and save_data does only createIfNotExist

But I could put updateOrCreate in a lib, it may become a usefull tool for others ?